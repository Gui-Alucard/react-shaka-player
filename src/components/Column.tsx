import React from 'react'
import styled from 'styled-components'

import Row from './Row'

type ChildrenProp = {
  children: React.ReactNode;
  position: string;
  height: string;
  top: number;
  left: number;
  alignItems: string;
  zIndex: number;
  style: any;
  opacity: number;
  ref: React.ForwardedRef<unknown>;
};

const Column = React.forwardRef((props: ChildrenProp, ref) => <Row flexDirection='column' ref={ref} {...props} />)

export const ColumnDesktop = styled(Column) <ChildrenProp>`
  display: none;
  @media (min-width: 1200px) {
    display: flex;
  }
`

export const ColumnMobile = styled(Column) <ChildrenProp>`
  display: flex;
  @media (min-width: 1200px) {
    display: none;
  }
`

export default Column
