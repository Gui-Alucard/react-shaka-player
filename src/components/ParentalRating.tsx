import React from 'react'
import PropTypes from 'prop-types'

import Row from './Row'
import Text from './Text'

const ParentalRatingComponent = ({
  parentalRating,
  size = 16,
  increaseSizeSelfClassified = false,
  textProps = {},
  selfClassified,
  ...props
}: any) => {
  return (
    <>
      {parentalRating && (
        <Row
          size={increaseSizeSelfClassified && selfClassified ? size + 5 : size}
          borderRadius={3}
          alignItems='center'
          justifyContent='center'
          bg={`parentalRating.${parentalRating}`}
          mx={8}
          {...props}
        >
          <Text
            variant={selfClassified ? 'little' : 'tiny'}
            lineHeight={1}
            fontFamily={'\'Urbanist\', \'Arial\', \'Helvetica Neue\', \'Helvetica\', sans-serif'}
            color={parentalRating === '10' || parentalRating == '12' ? '#000' : '#fff'}
            fontWeight={700}
            {...textProps}
          >
            {selfClassified && 'A'}
            {parentalRating}
          </Text>
        </Row>
      )}
    </>
  )
}
ParentalRatingComponent.propTypes = {
  textProps: PropTypes.object,
  size: PropTypes.number,
  parentalRating: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  selfClassified: PropTypes.bool,
  increaseSizeSelfClassified: PropTypes.bool
}

export default ParentalRatingComponent
