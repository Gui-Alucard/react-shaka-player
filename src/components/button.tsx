import React from 'react';
import { IPlayerProps } from '../types';
import { Unmute, Rewind, Foward } from '../icons';

const Button = (props: IPlayerProps) => {
  const { onClick, label } = props;

  const handleIcon = (label: string) => {
    if (label && label === 'unmute') {
      return (
        <div id={label} className={`sbt-theme-container-${label}`} >
          <button onClick={onClick}>
            <Unmute width={20} height={18.87} style={{ marginRight: '5px' }} /> <p>Ativar som</p>
          </button>
        </div>
      )
    } else if (label && label === 'foward') {
      return (
        <div id={label} className={`sbt-theme-container-${label}`} >
          <button onClick={onClick}>
            <Foward width={36} height={42} /> <p>Avan√ßar</p>
          </button>
        </div>
      )
    } else if (label && label === 'rewind') {
      return (
        <div id={label} className={`sbt-theme-container-${label}`} >
          <button onClick={onClick}>
            <Rewind width={36} height={41} /> <p>Retroceder</p>
          </button>
        </div>
      )
    }
  }

  return (
    <div id={label} className={`sbt-theme-container-${label}`} >
      <button onClick={onClick}>
        {label && handleIcon(label)}
      </button>
    </div>
  )
};

export { Button };
