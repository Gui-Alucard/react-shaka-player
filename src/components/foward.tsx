import React from 'react';
import { IPlayerProps } from '../types';

const FowardButton = (props: IPlayerProps) => {
  const { onClick, label } = props;

  const handleEnter = () => {
    const element = document.getElementsByClassName('shaka-controls-container');
    element.item(0).setAttribute('shown', 'true');
  }

  const handleLeave = () => {
    const element = document.getElementsByClassName('shaka-controls-container');
    element.item(0).removeAttribute('shown');
  }

  return (
    <div className='sbt-theme-foward-container' >
      <button type='button' onClick={onClick} onMouseEnter={handleEnter} onMouseLeave={handleLeave} ><p>{label && label}</p></button>
    </div>
  )
};

export { FowardButton };
