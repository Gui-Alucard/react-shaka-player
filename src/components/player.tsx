import React, { useRef } from 'react';
import * as Hooks from '../hooks';

import { IPlayerProps } from '../types';

const ReactPlayer = (props: IPlayerProps) => {
  const videoRef = useRef<HTMLVideoElement | null>(null);
  const uiContainerRef = useRef<HTMLDivElement | null>(null);

  const { player, ui, overlayClassName } = Hooks.usePlayer(videoRef, uiContainerRef, props);
  Hooks.usePlayerListener(player, props);
  Hooks.useUIListener(ui, player, props);
  Hooks.useStats(player, props);
  Hooks.useAds(ui, player, props);

  const {
    className,
    config,
    onFoward,
    onRewind,
    onBuffering,
    onClick,
    onLoad,
    onPlayerError,
    onStatsChange,
    onTimeUpdate,
    onUiInteraction,
    playerClassName,
    superConfig,
    uiConfig,
    adsRequest,
    adsTagUrl,
    startTime,
    unmute,
    muted,
    ...newProps
  } = props;

  const style = {
    maxWidth: "100%",
    width: "100%",
    maxHeight: "100vh",
    overflow: "hidden",
    cursor: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  };

  const unmuteClassName = superConfig !== 'STREAMING' ? 'sbt-theme-unmute-container' : 'sbt-theme-streaming-unmute-container';

  return (
    <div style={style} ref={uiContainerRef} className={overlayClassName}>
      <video
        ref={videoRef}
        className={props.playerClassName}
        style={style}
        autoPlay={false}
        muted={!!muted}
        {...newProps}
      />
      {unmute?.p && unmute?.onUnmute && (
        <div className={unmuteClassName}>
          <button type='button' onClick={unmute?.onUnmute}>
            <p>{unmute?.p && unmute?.p}</p>
          </button>
        </div>
      )}
      {ui && onFoward && (
        <div className='foward-container' >
          <button type='button' onClick={onFoward} />
        </div>
      )}
      {ui && onRewind && (
        <div className='rewind-container'>
          <button type='button' onClick={onRewind} />
        </div>
      )}
    </div>
  )
};

export { ReactPlayer };
