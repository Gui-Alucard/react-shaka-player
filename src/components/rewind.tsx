import { Player as ShakaPlayer } from "shaka-player/dist/shaka-player.ui";
import React, { useRef } from 'react';
import * as Hooks from '../hooks';

import { IPlayerProps } from '../types';
import { RewindIcon } from '../icons/rewindIcon';

interface IButtonRewind {
  player?: ShakaPlayer;
  props?: IPlayerProps;
}

const ButtonRewind = ({ player, props }: IButtonRewind) => {
  const buttonRewind = useRef<HTMLButtonElement | null>(null);

  Hooks.useButton(player, props, buttonRewind);

  return (
    <div id='rewind' className='sbt-container-rewind'>
      <button ref={buttonRewind} onClick={props && props.onRewind}>
        <RewindIcon width={36} height={41} style={{ marginRight: '5px' }} /> <p>{props && props.label}</p>
      </button>
    </div>
  )
};

export { ButtonRewind };