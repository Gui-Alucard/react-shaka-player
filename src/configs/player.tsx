const vodConfig = {
  ui: {
    addSeekBar: true,
    doubleClickForFullscreen: true,
    enableFullscreenOnRotation: true,
    clearBufferOnQualityChange: true,
    enableTooltips: false,
    enableKeyboardPlaybackControls: false,
    forceLandscapeOnFullscreen: false,
    singleClickForPlayAndPause: false,
    addBigPlayButton: false,
    customContextMenu: false,
    controlPanelElements: [
      'play_pause',
      'mute',
      'volume',
      'time_and_duration',
      'spacer',
      'fullscreen',
      'quality'
    ],
    volumeBarColors: {
      base: '#ffffff4c',
      level: '#0060ad'
    },
    seekBarColors: {
      base: '#ffffff4c',
      buffered: '#ffffff4c',
      played: '#e93035',
    }
  },
};

const streamingConfig = {
  player: {
    manifest: {
      dash: {
        ignoreMinBufferTime: true,
      },
    },
    streaming: {
      lowLatencyMode: true,
      inaccurateManifestTolerance: 0,
      rebufferingGoal: 1,
      durationBackoff: 0
    },
  },
  ui: {
    addSeekBar: false,
    doubleClickForFullscreen: true,
    enableFullscreenOnRotation: true,
    enableKeyboardPlaybackControls: false,
    singleClickForPlayAndPause: false,
    forceLandscapeOnFullscreen: false,
    clearBufferOnQualityChange: false,
    addBigPlayButton: false,
    customContextMenu: false,
    controlPanelElements: ['mute', 'volume', 'fullscreen', 'quality', 'picture_in_picture'],
    volumeBarColors: {
      base: '#ffffff4c',
      level: '#0060ad'
    },
    seekBarColors: {
      base: '#ffffff4c',
      buffered: '#ffffff4c',
      played: '#e93035',
    }
  },
};

export { vodConfig, streamingConfig };
