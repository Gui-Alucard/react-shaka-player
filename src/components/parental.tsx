import React from 'react';
import { IPlayerProps } from '../types';
import Column from './Column';
import Row from './Row';
import Text from './Text';
import ParentalRating from './ParentalRating';

const Parental = (props: IPlayerProps) => {
  const { content } = props

  return (
    <div>
      {content && !content.isLive && content?.episode?.parental_rating && (
        <Column
          position='fixed'
          height={content.isIphone ? '-webkit-fill-available' : 'fit-content'}
          top={30}
          left={30}
          alignItems='center'
          zIndex={3}
          style={{ pointerEvents: content.showControls && content.showParentalRating ? 'none' : 'auto' }}
          opacity={!content.showControls && content.showParentalRating ? 1 : 0}
          {...props}
        >
          <Row alignItems='center' width='100%' mt={42} px={46}>
            <Text variant='small' textShadow='0px 1px 2px black' borderLeft='2px solid #ef4846' px={8}>
              Classificação
            </Text>
            <ParentalRating
              parentalRating={content?.episode?.parental_rating}
              selfClassified={content?.episode?.self_classified}
              size={25}
              increaseSizeSelfClassified={true}
              textProps={{ variant: 'small', fontWeight: 'bold', fontFamily: '\'Urbanist\', \'Arial\', \'Helvetica Neue\', \'Helvetica\', sans-serif' }}
            />
          </Row>
          <Row alignItems='center' width='100%' mt={8} px={46}>
            <Text variant='tiny' textShadow='0px 1px 2px black' pl={10}>
              {content?.episode?.parental_rating_justification}
            </Text>
          </Row>
        </Column>
      )}
    </div>
  )
};

export { Parental };
