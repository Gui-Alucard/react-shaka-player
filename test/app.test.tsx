import * as Enzyme from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { ReactPlayer as Player } from '../src/components/player';
import { Button as Button } from '../src/components/button';

Enzyme.configure({ adapter: new Adapter() });

describe("Player", () => {
  it("Should render correctly", () => {
    const tree = renderer.create(<Player />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should renders source[manifest] string correctly", () => {
    const props = { src: "this is sample source" };
    const tree = Enzyme.shallow(<Player {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("Should render className container correctly", () => {
    const props = { className: "sbt-theme" };
    const tree = Enzyme.shallow(<Player {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("Should render playerClassName container correctly", () => {
    const props = { playerClassName: "player-class-name" };
    const tree = Enzyme.shallow(<Player {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("Should render container className and playerClassName container correctly", () => {
    const props = {
      className: "sbt-theme",
      playerClassName: "player-class-name",
    };
    const tree = Enzyme.shallow(<Player {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe("Button", () => {
  it("Should render correctly", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should renders Button Unmute correctly", () => {
    const props = { label: "unmute" };
    const tree = Enzyme.shallow(<Button {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("Should renders Button Foward correctly", () => {
    const props = { label: "foward" };
    const tree = Enzyme.shallow(<Button {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("Should renders Button Rewind correctly", () => {
    const props = { label: "rewind" };
    const tree = Enzyme.shallow(<Button {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("Should render the onclick function correctly", () => {
    const props = { onClick: () => true };
    const tree = Enzyme.shallow(<Button {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
