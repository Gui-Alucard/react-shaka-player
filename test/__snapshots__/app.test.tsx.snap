// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render correctly 1`] = `
<div
  className="sbt-theme-container-undefined"
>
  <button />
</div>
`;

exports[`Button Should render the onclick function correctly 1`] = `ShallowWrapper {}`;

exports[`Button Should renders Button Foward correctly 1`] = `ShallowWrapper {}`;

exports[`Button Should renders Button Rewind correctly 1`] = `ShallowWrapper {}`;

exports[`Button Should renders Button Unmute correctly 1`] = `ShallowWrapper {}`;

exports[`Player Should render className container correctly 1`] = `ShallowWrapper {}`;

exports[`Player Should render container className and playerClassName container correctly 1`] = `ShallowWrapper {}`;

exports[`Player Should render correctly 1`] = `
<div
  className="sbt-theme"
  style={
    {
      "alignItems": "center",
      "cursor": "none",
      "display": "flex",
      "justifyContent": "center",
      "maxHeight": "100vh",
      "maxWidth": "100%",
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <video
    autoPlay={true}
    muted={true}
    style={
      {
        "alignItems": "center",
        "cursor": "none",
        "display": "flex",
        "justifyContent": "center",
        "maxHeight": "100vh",
        "maxWidth": "100%",
        "overflow": "hidden",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Player Should render playerClassName container correctly 1`] = `ShallowWrapper {}`;

exports[`Player Should renders source[manifest] string correctly 1`] = `ShallowWrapper {}`;
